sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -1016,6 +1016,8 @@ class FilterPushdownSuite extends PlanTest {
     val correctAnswer = x.where("x.a".attr === 5).join(y.where("y.a".attr === 5),
         condition = Some("x.a".attr === Rand(10) && "y.b".attr === 5))
 
-    comparePlans(Optimize.execute(originalQuery.analyze), correctAnswer.analyze)
```

Comment:
The reason I don't use `Project` operator is that might change the result correctness, for example, suppose we have join condition `rand(1) > 0 &&  rand(11) < 0`, to use `Project` operator, we will always eval the both rand functions, which will change the result.